File: src/main/java/com/instaimg/crawl/InstagramFX.java
package com.instaimg.crawl;

import javafx.application.Application;
import javafx.fxml.FXMLLoader;
import javafx.scene.Parent;
import javafx.scene.Scene;
import javafx.stage.Stage;

import java.util.Objects;

public class InstagramFX  extends Application {
    public static void main(String[] args) {
        launch(args);
    }

    @Override
    public void start(Stage primaryStage) throws Exception {
        Parent root = FXMLLoader.load(Objects.requireNonNull(getClass().getResource("instagramFX.fxml")));
        primaryStage.setTitle("Instagram Downloader");
        Scene scene = new Scene(root, 800, 600);
        scene.getStylesheets()
                .add(Objects.requireNonNull(getClass().getClassLoader().getResource("css/style.css")).toExternalForm());
        primaryStage.setScene(scene);
        primaryStage.show();
    }
}

File: src/main/kotlin/com/instaimg/crawl/InstagramFX.kt
package com.instaimg.crawl

import javafx.application.Application
import javafx.fxml.FXMLLoader
import javafx.scene.Parent
import javafx.scene.Scene
import javafx.scene.text.Font
import javafx.stage.Stage

class InstagramFX : Application() {

    override fun start(primaryStage: Stage) {
        Font.loadFont(
            checkNotNull(javaClass.classLoader.getResourceAsStream("font/Raleway-VariableFont_wght.ttf")) {
                "Cannot find Raleway font on classpath"
            }, 24.0
        )
        val root = FXMLLoader.load<Parent>(
            checkNotNull(javaClass.getResource("instagramFX.fxml")) {
                "Cannot find instagramFX.fxml on classpath"
            }
        )
        primaryStage.title = "Instagram Downloader"
        val scene = Scene(root, 800.0, 600.0)
        scene.stylesheets.add(
            checkNotNull(javaClass.classLoader.getResource("css/style.css")) {
                "Cannot find css/style.css on classpath"
            }.toExternalForm()
        )
        primaryStage.scene = scene
        primaryStage.show()
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            launch(InstagramFX::class.java, *args)
        }
    }
}
